iquip/apps/scheduler.py

```diff
@@ -1,14 +1,95 @@
 """App module for showing the scheduled queue for experiments."""
 
 import enum
-from typing import Any, Optional, Tuple
+import logging
+from typing import Any, Callable, Iterable, Optional, Tuple
 
-from PyQt5.QtCore import QAbstractTableModel, QModelIndex, QObject, Qt, QVariant
+import requests
+from PyQt5.QtCore import (
+    pyqtSignal, pyqtSlot, QAbstractTableModel, QModelIndex, QObject, Qt, QThread, QVariant
+)
 from PyQt5.QtWidgets import QTableView, QVBoxLayout, QWidget
 
 import qiwis
 from iquip.protocols import SubmittedExperimentInfo
 
+logger = logging.getLogger(__name__)
+
+
+class _ScheduleThread(QThread):
+    """QThread for obtaining the current scheduled queue from the proxy server.
+    
+    Signals:
+        fetched(isChanged, updatedTime, schedule): The current scheduled queue is fetched.
+          The "schedule" is a list with SubmittedExperimentInfo elements.
+          The "updatedTime" is the time when the fetched schedule was updated.
+          If a timeout occurs, i.e. the queue is not changed, the "isChanged" is set to False.
+
+    Attributes:
+        ip: The proxy server IP address.
+        port: The proxy server PORT number.
+        updatedTime: The last updated time, in the format of time.time().
+    """
+
+    fetched = pyqtSignal(bool, float, list)
+
+    def __init__(
+        self,
+        ip: str,
+        port: int,
+        updatedTime: Optional[float],
+        callback: Callable[[bool, float, Iterable[SubmittedExperimentInfo]], None],
```

Comment:
Since the third argument of `fetched` is `list`, you should require `callback` to accept `List` rather than `Iterable`.
You are imposing too strong condition (it is enough for `callback` to accept lists, not tuples, etc.).

Note that `Callable[[List], None]` _contains_ `Callable[[Iterable], None]`. Don't be confused!

Reply:
I have a question! Why does the former contain the latter? I think `Iterable` contains `List`.